jobs/validate/calico-spec

```diff
@@ -88,13 +93,19 @@ function juju::deploy
          --force \
          --channel "$JUJU_DEPLOY_CHANNEL" "$JUJU_DEPLOY_BUNDLE"
 
-    if [[ "$ROUTING_MODE" = bgp* ]]; then
-      python $WORKSPACE/jobs/integration/tigera_aws.py disable-source-dest-check
-      python $WORKSPACE/jobs/integration/tigera_aws.py assign-ipv6-addrs
-    fi
+    (
+      if [[ "$ROUTING_MODE" = bgp* ]]; then
+        python $WORKSPACE/jobs/integration/tigera_aws.py disable-source-dest-check
+        python $WORKSPACE/jobs/integration/tigera_aws.py assign-ipv6-addrs
```

Comment:
Looks like errors from these are still ignored, since the subshell will only return the code of the last command run within it. e.g.

```
$ (false; true)
$ echo $?
0
```

Maybe a `set -e` in the subshell would do what's intended here?

```
(
  set -e
  ...
)
ret=$?
```